nightfall-client/src/services/finalise-withdrawal.mjs

```diff
@@ -16,7 +16,9 @@ export async function finaliseWithdrawal(transactionHash) {
   const transactions = await Promise.all(
     block.transactionHashes.map(t => getTransactionByTransactionHash(t)),
   );
-  const index = transactions.findIndex(f => f.transactionHash === transactionHash);
+
+  // block can contain multiple transactions
+  const index = transactions.findIndex(f => transactionHash === f?.transactionHash ?? '0x0');
```

Comment:
I tried to reproduce the bug locally but couldn't: I made a few transfers and a withdrawal > made block > finalised withdrawal successfully.

By looking at the proposed fix, I suspect we are trying to resolve another problem: you cannot access `transactionHash` in `f` if such `f` is null.

The array `transactions` will contain a `null` element if you do not have a particular transaction in the `transactions` collection - to get this far in the service, it means that you have the block in `blocks` collection but this block contains transactions from other clients (not only your transaction).

So the bug is that a withdrawal can't be finalised when a block contains transactions that are not in your database :nerd_face:

Reply:
thanks @imagobea. You are correct in your analysis. I will modify the test to check for this condition